root/types.js

```diff
@@ -1,20 +1,49 @@
-// @flow
-// Copyright (C) 2017 MetaBrainz Foundation
-//
-// This file is part of MusicBrainz, the open internet music database,
-// and is licensed under the GPL version 2, or (at your option) any
-// later version: http://www.gnu.org/licenses/gpl-2.0.txt
+/*
+ * @flow
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
 
 
-// Types are in alphabetical order.
-//
-// The definitions in this file are intended to model the output of the
-// TO_JSON methods under lib/MusicBrainz/Server/Entity/; those are precisely
-// how data is serialized for us.
+/*
+ * Types are in alphabetical order.
+ *
+ * The definitions in this file are intended to model the output of the
+ * TO_JSON methods under lib/MusicBrainz/Server/Entity/; those are precisely
+ * how data is serialized for us.
+ */
+
+declare type AreaT = CoreEntityT & {
+  typeName: string;
```

Comment:
From what I see in lib/MusicBrainz/Server/Entity/Role/Type.pm, *all* of the `typeName` properties should be typed as `typeName?: string` (meaning they could be missing).

It's important the types exactly model what the `TO_JSON` gives us, otherwise the type checking won't be sound and we could introduce logical errors.